src/main/java/com/dabsquared/gitlabjenkins/gitlab/api/impl/V4GitLabClientBuilder.java

```diff
@@ -1,24 +1,33 @@
 package com.dabsquared.gitlabjenkins.gitlab.api.impl;
 
-import com.dabsquared.gitlabjenkins.gitlab.api.model.MergeRequest;
+import com.dabsquared.gitlabjenkins.gitlab.api.GitLabClientBuilder;
 import hudson.Extension;
-import java.util.function.Function;
+import org.gitlab4j.api.GitLabApi;
+import org.gitlab4j.api.GitLabApi.ApiVersion;
+import org.gitlab4j.api.GitLabApiException;
 import org.kohsuke.accmod.Restricted;
 import org.kohsuke.accmod.restrictions.NoExternalUse;
 
 @Extension
 @Restricted(NoExternalUse.class)
-public final class V4GitLabClientBuilder extends ResteasyGitLabClientBuilder {
-    private static final int ORDINAL = 1;
-    private static final Function<MergeRequest, Integer> MERGE_REQUEST_ID_PROVIDER =
-            new Function<MergeRequest, Integer>() {
-                @Override
-                public Integer apply(MergeRequest mergeRequest) {
-                    return mergeRequest.getIid();
-                }
-            };
+public final class V4GitLabClientBuilder extends GitLabClientBuilder {
 
     public V4GitLabClientBuilder() {
-        super(V4GitLabApiProxy.ID, ORDINAL, V4GitLabApiProxy.class, MERGE_REQUEST_ID_PROVIDER);
+        super("V4", 1);
+    }
+
+    @Override
+    public GitLabApi buildClient(
+            String url, String token, boolean ignoreCertificateErrors, int connectionTimeout, int readTimeout) {
+        GitLabApi client = null;
+        try {
+            client = new GitLabApi(ApiVersion.V4, url, token);
+            client.getUserApi().getCurrentUser();
+            client.setIgnoreCertificateErrors(ignoreCertificateErrors);
+            client.setRequestTimeout(connectionTimeout, readTimeout);
+            return client;
+        } catch (GitLabApiException e) {
+            return null;
```

Comment:
The abstract `com.dabsquared.gitlabjenkins.gitlab.api.GitLabClientBuilder#buildClient` method is annotated with `@NonNull`, so this is in violation of the API contract. SpotBugs correctly raises a warning.